### ISSUE TITLE
WYSIWYG forum editor

### ISSUE THREAD

**Issue by RedGuy12** [ScratchAddons/ScratchAddons#1442]
(I understand this would be rather hard! Just putting the idea out there!)

Some people might not understand Bbcode so a WYSIWYG (what you see is what you get) editor for the forums world be nice.

**Comment by Secret-chest** [comment]
too complicated

**Comment by Secret-chest** [comment]
you can just click preview to see the result

**Comment by MiniCoder11** [comment]
> you can just click preview to see the result

That creates a new solution. What if instead of a complicated addon to the forum editor box, we simply refresh the preview pane repeatedly to achieve the same result of a live preview (may become slow with large scripts)?

**Comment by mxmou** [comment]
> That creates a new solution. What if instead of a complicated addon to the forum editor box, we simply refresh the preview pane repeatedly to achieve the same result of a live preview (may become slow with large scripts)?

I've thought of that before, and it would be really useful - but previewing currently means a request to the server, which can be slow. A solution for that would be parsing BBCode on the client, but that means the preview wouldn't be 100% accurate.

**Comment by Hans5958** [comment]
> too complicated

Just because it is too complicated, doesn't mean it is impossible.

**Comment by MiniCoder11** [comment]
Would a good compromise be to refresh the preview pane whenever the user stops typing for a certain amount of time (probably about a second or so)?

**Comment by mxmou** [comment]
> Would a good compromise be to refresh the preview pane whenever the user stops typing for a certain amount of time (probably about a second or so)?

That would work. As I said, the preview could be faster if it was client-side - maybe that could be a setting?

**Comment by MiniCoder11** [comment]
> That would work. As I said, the preview could be faster if it was client-side - maybe that could be a setting?

That would a be great. Would we be able to implement the new 3.0 Scratch blocks as part of the client-side parsing?

**Comment by mxmou** [comment]
> > That would work. As I said, the preview could be faster if it was client-side - maybe that could be a setting?
> 
> That would a be great. Would we be able to implement the new 3.0 Scratch blocks as part of the client-side parsing?

Yes, but it doesn't make much sense to have 3.0 blocks in the preview if they don't appear in the actual post, and the latter is harder to implement.

**Comment by MiniCoder11** [comment]
> Yes, but it doesn't make much sense to have 3.0 blocks in the preview if they don't appear in the actual post, and the latter is harder to implement.

I overlooked that part. Consistency is probably more important then introducing the new blocks. On a side note, are there any addons planned to add the "<() contains ()?> and "costume [v name]" blocks to the forum block picker? Currently we have to format the blocks ourselves every time we want to use them.

**Comment by RedGuy12** [comment]
> On a side note, are there any addons planned to add the "<() contains ()?> and "costume [v name]" blocks to the forum block picker? Currently we have to format the blocks ourselves every time we want to use them.

Create a new issue
________________________________

**Comment by Samq64** [comment]
Implemented in #4253?

**Comment by Secret-chest** [comment]
> (I understand this would be rather hard! Just putting the idea out there!)
> 
> Some people might not understand Bbcode so a WYSIWYG (what you see is what you get) editor for the forums world be nice.

The current one is a WYWIWYG. What you want is what you get.

**Comment by apple502j** [comment]
Reposting here;

> Also, WYSIWYG is NOT the live preview. For that you should ideally be able to click on the preview, type on the preview, then the textbox should be changed to reflect it - no need to type BBCode at all.
